core/Enclave/enclave_hsm.cpp

```diff
@@ -489,30 +532,44 @@ sgx_status_t enclave_rsa_decrypt(const uint8_t *cmk_blob, size_t cmk_blob_size,
 {
     sgx_status_t ret = SGX_ERROR_UNEXPECTED;
 
-    rsa_key_data_t *temp = (rsa_key_data_t*)cmk_blob;
-
-    BIO *bio = BIO_new_mem_buf(temp->rsa_private_key, -1);
+    // load private key
+    rsa_key_data_t *temp_pri_kry = (rsa_key_data_t*)cmk_blob;
+    BIO *bio = BIO_new_mem_buf(temp_pri_kry->rsa_private_key, -1);
     RSA *rsa = PEM_read_bio_RSAPrivateKey(bio, NULL, NULL, g_domain_key);
     
-
-    if(rsa == NULL)
-    {
+    if (rsa == NULL) {
         printf("failed to load rsa key\n");
+
+        RSA_free(rsa);
+        BIO_free(bio);
+        temp_pri_kry = NULL;
+
         return SGX_ERROR_INVALID_PARAMETER;
     }
 
-    if (plaintext_len == 0)
-    {
-        unsigned char* to;
-        RSA_private_decrypt(ciphertext_len, ciphertext, to, rsa, RSA_PKCS1_OAEP_PADDING);
-        *req_plaintext_len = strlen((char*)to);
-    }
-    else
-    {    
-        RSA_private_decrypt(ciphertext_len, ciphertext, plaintext, rsa, RSA_PKCS1_OAEP_PADDING);
+    int res;
+    if (plaintext_len == 0) {
+        // compute plaintext length
+        unsigned char to[RSA_size(rsa)]; 
+        
+        res = RSA_private_decrypt(ciphertext_len, ciphertext, to, rsa, padding_mode);        
```

Comment:
What does res mean?
RSA_private_decrypt() returns the size of the recovered plaintext.